eos/utils/expression-observable.hh

```diff
@@ -0,0 +1,193 @@
+/* vim: set sw=4 sts=4 et foldmethod=syntax : */
+
+/*
+ * Copyright (c) 2010, 2011, 2015, 2016, 2017 Danny van Dyk
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_SRC_UTILS_EXPRESSION_OBSERVABLE_HH
+#define EOS_GUARD_SRC_UTILS_EXPRESSION_OBSERVABLE_HH 1
+
+#include <eos/observable-impl.hh>
+#include <eos/utils/expression-cloner.cc>
+#include <eos/utils/expression-evaluator.cc>
+#include <eos/utils/expression-fwd.hh>
+#include <eos/utils/expression-maker.cc>
+#include <eos/utils/expression-parser-impl.hh>
+#include <eos/utils/expression-printer.cc>
+#include <eos/utils/wrapped_forward_iterator-impl.hh>
+
+
+namespace eos
+{
+    using eos::exp::Expression;
+
+    class ExpressionObservable :
+        public Observable
+    {
+        public:
+
+        private:
+            QualifiedName _name;
+
+            Parameters _parameters;
+
+            Kinematics _kinematics;
+
+            Options _options;
+
+            Expression _expression;
+
+        public:
+            ExpressionObservable(const QualifiedName & name,
+                    const Parameters & parameters,
+                    const Kinematics & kinematics,
+                    const Options & options,
+                    const Expression & expression) :
+                _name(name),
+                _parameters(parameters),
+                _kinematics(kinematics),
+                _options(options),
+                _expression(expression)
+            {
+                // Uses to be implemented
```

Comment:
Should we make a ``ParameterUser`` a ``ExpressionVisitor``? That would be an elegant solution to implementing parameter usage.